evaluator.py

```diff
@@ -0,0 +1,151 @@
+                pnls: a list-like objects of floats.
+                positions: a list-like of dict-like objects.
+        """
+        dates = [item.date for item in data]
+        pnls = np.array([item.pnl for item in data])
+        positions_raws = [item.positions_raw for item in data]
+        positions_nps = [item.positions_np for item in data]
+        serialized = (
+            lambda var: Array(ctypes.c_char, pickle.dumps(var), lock=False))
+        packed = {
+            DATES: serialized(dates),
+            PNLS: serialized(pnls),
+            POSITIONS_RAW: serialized(positions_raws),
+            POSITIONS_NP: serialized(positions_nps)
+        }
+        processes = []
+        results_queue = Queue()
+        for indicator in self.indicators:
+            process = Process(
+                target=evaluate,
+                args=(results_queue, indicator),
+                kwargs=packed)
+            process.start()
+            processes.append(process)
+        for process in processes:
+            process.join()
+        results = {}
+        while not results_queue.empty():
+            indicator_name, result = results_queue.get()
+            results[indicator_name] = result
+        return results
+
+def evaluate(results, indicator, **serialized):
+    """
+    Decode all shared variables into evalution function inputs, and return
+    their results by results (a multiprocess.Queue).
+    """
+    results.put((indicator.__name__, indicator(**serialized)))
+
+def get_all_indicators():
+    """
+    Return a list of all available built-in indicators.
+    """
+    return list(indicators_all.keys())
+
+def default(func):
+    """
+    Set func to be a default indicators.
+    """
+    indicators_default.append(func)
+    return func
+
+def inputs(*needed_fields):
+    """
+    Decorator for eval_functions to extract their inputs from encoded data.
+    Only fields in needed_fields will be decoded to improve the performance.
+
+    All and only functions decorated by this will be considered as indicators.
+    They will be registered into indicator_list for lookup.
+    """
+    def middle(func):
+        def final(**available_fileds):
+            return func(*(
+                pickle.loads(available_fileds[field])
+                for field in needed_fields))
+        final.__name__ = func.__name__
+        # register the function into indicators_all
+        indicators_all[final.__name__] = final
+        return final
+    return middle
+
+@default
+@inputs(PNLS)
+def returns(pnls):
+    """
+    Average annual returns
+    """
+    return np.mean(pnls) * 252
+
+@default
+@inputs(PNLS)
+def sharpe(pnls):
+    """
+    Sharpe ratio
+    """
+    return np.mean(pnls) / np.std(pnls)
+
+@default
+@inputs(POSITIONS_NP)
+def turnover(positions_nps):
+    """
+    Turnover rate
+    """
+    d_turnover = np.empty(len(positions_nps)-1) # daily turnover rate
+    d_change = abs(positions_nps[1:] - positions_nps[:-1])
```

Comment:
Could you use more meaningful word for _"d_"_ in _"d_turnover"_ and _"d_change"_ ? It made me confused what "d" means.